joblib/memory.py

```diff
@@ -523,7 +525,6 @@ def _cached_call(self, args, kwargs):
                           '(args=%s, kwargs=%s)\n %s' %
                           (args, kwargs, traceback.format_exc()))
 
-                shutil.rmtree(output_dir, ignore_errors=True)
```

Comment:
This is one of the possibly controversial change. Whereas before we would remove the entire cache folder, we now keep the possibly corrupted 'output.pkl' knowing that he will be overwritten eventually. According to @GaelVaroquaux and @ogrisel IRL it seems like an acceptable choice.

Removing the folder is actually likely to add additional recomputations.